test/JsmnSolLib.ts

```diff
@@ -0,0 +1,466 @@
+        });
+
+        it("should parse in 0x800-0xffff range", async () => {
+            const jsmnSolLib = await deployParser(wallet);
+            const json = '{"\xe0\xa0\x80": "\xe0\xa0\x81", "\xef\xbf\xbe": "\xef\xbf\xbf"}';
+            const [returnValue, tokens, actualNum] = await jsmnSolLib.parse(json, 10);
+
+            expect(returnValue).to.eq(0, "Should be valid json");
+            expect(actualNum).to.eq(5, "Should have 1 object and 2 key value pairs");
+            expect(await jsmnSolLib.getBytes(json, tokens[1].start, tokens[1].end)).to.eq("\xe0\xa0\x80", "failed to parse start of range");
+            expect(await jsmnSolLib.getBytes(json, tokens[2].start, tokens[2].end)).to.eq("\xe0\xa0\x81", "failed to parse within range start");
+            expect(await jsmnSolLib.getBytes(json, tokens[3].start, tokens[3].end)).to.eq("\xef\xbf\xbe", "failed to parse within range end");
+            expect(await jsmnSolLib.getBytes(json, tokens[4].start, tokens[4].end)).to.eq("\xef\xbf\xbf", "failed to parse end of range");
+        });
+
+        it("should parse in 0x10000-0x10ffff range", async () => {
+            const jsmnSolLib = await deployParser(wallet);
+            const json = '{"\xf0\x90\x80\x80": "\xf0\x90\x80\x81", "\xf4\x8f\xbf\xbe": "\xf4\x8f\xbf\xbf"}';
+            const [returnValue, tokens, actualNum] = await jsmnSolLib.parse(json, 10);
+
+            expect(returnValue).to.eq(0, "Should be valid json");
+            expect(actualNum).to.eq(5, "Should have 1 object and 2 key value pairs");
+            expect(await jsmnSolLib.getBytes(json, tokens[1].start, tokens[1].end)).to.eq("\xf0\x90\x80\x80", "failed to parse start of range");
+            expect(await jsmnSolLib.getBytes(json, tokens[2].start, tokens[2].end)).to.eq("\xf0\x90\x80\x81", "failed to parse within range start");
+            expect(await jsmnSolLib.getBytes(json, tokens[3].start, tokens[3].end)).to.eq("\xf4\x8f\xbf\xbe", "failed to parse within range end");
+            expect(await jsmnSolLib.getBytes(json, tokens[4].start, tokens[4].end)).to.eq("\xf4\x8f\xbf\xbf", "failed to parse end of range");
+        });
+
+        it("should accept unicode encoding", async () => {
+            const jsmnSolLib = await deployParser(wallet);
+            const json = '{"\uD834\uDD1E": "👁️"}';
+            const [returnValue, tokens, actualNum] = await jsmnSolLib.parse(json, 10);
+
+            expect(await jsmnSolLib.getBytes(json, tokens[1].start, tokens[1].end)).to.eq("\uD834\uDD1E", "failed to parse g clef");
+            expect(await jsmnSolLib.getBytes(json, tokens[2].start, tokens[2].end)).to.eq("👁️", "failed to parse a single emoji character");
+            expect(returnValue).to.eq(0, "Should be valid json");
+            expect(actualNum).to.eq(3, "Should have 1 object and 1 key value pair");
+        });
+    });
+
+    describe.skip("file parsing", () => {
```

Comment:
This is currently skipping the json file tests
```suggestion
    describe("file parsing", () => {
```

Reply:
So about 20% of the file tests currently fail. I started going down the route of making a 100% standards compliant json parser in Solididy and decided that's not what this project is about and doesn't change how we parse client data that's signed and generated by the browser. None of the failed tests let you do things like bypass the validation, they just allow json formats that aren't valid (like trailing commas and missing brackets). 

Would it be better to not include the compliance file tests at all if we can't pass them all or is it helpful to know which ones fail that don't look that serious?